Firestore/core/test/firebase/firestore/remote/datastore_test.cc

```diff
@@ -123,6 +126,8 @@ class DatastoreTest : public testing::Test {
     if (!is_shut_down) {
       Shutdown();
     }
+    // Ensure that nothing remains on the AsyncQueue before destroying it.
+    worker_queue->EnqueueBlocking([] {});
```

Comment:
Note: we use this pattern so much, I wonder if we should create a test-only helper function like `DrainWorkerQueue`.